code/builders/builder-vite/src/codegen-modern-iframe-script.ts

```diff
@@ -21,9 +21,12 @@ export async function generateModernIframeScriptCode(options: Options, projectRo
   // and the HMR handler.  We don't use the hot.accept callback params because only the changed
   // modules are provided, the rest are null.  We can just re-import everything again in that case.
   const getPreviewAnnotationsFunction = `
-  const getProjectAnnotations = async () => {
+  const getProjectAnnotations = async (hmrPreviewAnnotationModules = []) => {
     const configs = await Promise.all([${previewAnnotationURLs
-      .map((previewAnnotation) => `import('${previewAnnotation}')`)
+      .map(
+        (previewAnnotation, index) =>
+          `hmrPreviewAnnotationModules.at(${index}) ?? import('${previewAnnotation}')`
```

Comment:
This is a little hard to grok without the benefit of the explanation in your PR description, I wonder if it would be helpful to have a brief comment describing why this is necessary.